scripts/lib/CIME/namelist.py

```diff
@@ -1414,6 +1453,18 @@ def _parse_variable_name(self, allow_equals=True):
         if not is_valid_fortran_name(text_check):
             raise _NamelistParseError("%r is not a valid variable name at %s" %
                                       (str(text), self._line_col_string()))
+
+        # a particular value of a namelist array may be assigned using syntax
+        # foo(2) = 7
+
+        nindex = fortran_name_index(text_check)
+        if nindex is not None:
+            text = text_check[:text.find('(')]
+            vname = text.lower()
+            if vname in self._nameindex:
+                self._nameindex[vname].append(nindex)
+            else:
+                self._nameindex[vname] = [nindex]
```

Comment:
If `fortran_name_index` returned a list of integers instead of a single int, you could always use extend.
Before you work on that, be sure you have figured out how to handle sections correctly.
For instance, `foo(2::2)` is valid as `foo(::)` or `foo(8,4,-1`.
Another issue with trying to insert values in lists (as opposed to just listing each section in order) is that you do now know the bounds of the actual array. It is perfectly valid for an array in a namelist to have negative bounds (e.g., `integer :: foo(-3:3)`). Not knowing those bounds, how will you process:
```
foo(:) = 16*'a'
foo(-3) = 'b'
```